src/hybrid_failure_discovery/envs/conveyorbelt_env.py

```diff
@@ -0,0 +1,247 @@
+"""This module defines the ConveyorBeltEnv environment for simulating a 1D
+conveyor belt with loading and shifting actions."""
+
+from dataclasses import dataclass
+
+import matplotlib.pyplot as plt
+import numpy as np
+from gymnasium.core import RenderFrame
+from numpy.typing import NDArray
+from tomsutils.spaces import EnumSpace
+from tomsutils.utils import fig2data
+
+from hybrid_failure_discovery.envs.constraint_based_env_model import (
+    ConstraintBasedGymEnv,
+)
+
+
+@dataclass(frozen=True)
+class ConveyorBeltState:
+    """Represents the current state of the conveyor belt, including box
+    values."""
+
+    values: NDArray[np.float32]  # Vector of 6 float32s representing the state
+
+
+@dataclass(frozen=True)
+class ConveyorBeltAction:
+    """Represents a discrete action taken in the conveyor belt environment."""
+
+    index: int  # Discrete action index from 0 to 2
+
+
+class ConveyorBeltEnv(ConstraintBasedGymEnv[ConveyorBeltState, ConveyorBeltAction]):
+    """A simplified 1D conveyor belt environment with load/unload behavior and
+    3 actions."""
+
+    metadata = {"render_modes": ["rgb_array"], "render_fps": 10}
+
+    def __init__(self, render_mode: str = "rgb_array", seed: int = 0) -> None:
+        self.render_mode = render_mode
+        self.observation_dim = 6
+        self._last_inserted_value: float | None = None
+        self.action_dim = 3  # Now only actions 0, 1, 2 are valid
+        super().__init__(seed)
+
+        self._step_count = 0
+
+    def _create_action_space(self):
```

Comment:
let's type annotate the output. also, when we say "space" in this repo, we usually mean some subclass of `gymnasium.Space`, such as a `Box` space. in this case, maybe we want to use a `Discrete` space. see here: https://gymnasium.farama.org/api/spaces/fundamental/#gymnasium.spaces.Discrete